api/v1/system_webhook.go

```diff
@@ -146,7 +146,8 @@ func validateCertificates(obj *System) error {
 			// - Openstack_CA/OpenLDAP/Docker/SSL(HTTPS)
 			if c.Type == OpenstackCACertificate || c.Type == OpenLDAPCertificate ||
 				c.Type == DockerCertificate || c.Type == PlatformCertificate {
-				continue
+				msg := fmt.Sprintf("Unexpected certificate type: %s", c.Type)
+				return errors.New(msg)
```

Comment:
I would prefer to use kubebuilder enum validation for this instead but we cannot make this change until we have an answer for behavior of the DM when certs such as openstack_ca / docker / openldap / platform are applied in version N-1 and upgraded to N.